tests/integration-tests/tests/common/utils.py

```diff
@@ -40,21 +46,43 @@
     "ubuntu1804": {"name": "Deep Learning Base AMI (Ubuntu 18.04)*", "owners": ["amazon"]},
 }
 
-AMI_TYPE_DICT = {"official": OS_TO_OFFICIAL_AMI_NAME_OWNER_MAP, "remarkable": OS_TO_REMARKABLE_AMI_NAME_OWNER_MAP}
+OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP = {
+    "alinux": {"name": "amzn-hvm-x86_64-*", "owners": ["amazon"]},
+    "alinux2": {"name": "amzn2-hvm-*-*", "owners": ["amazon"]},
+    "centos6": {"name": "centos6-hvm-x86_64-*", "owners": ["amazon"]},
+    "centos7": {"name": "centos7-hvm-x86_64-*", "owners": ["amazon"]},
+    "ubuntu1604": {"name": "ubuntu-1604-lts-hvm-x86_64-*", "owners": ["amazon"]},
+    "ubuntu1804": {"name": "ubuntu-1804-lts-hvm-*-*", "owners": ["amazon"]},
+}
+
+AMI_TYPE_DICT = {
+    "official": OS_TO_OFFICIAL_AMI_NAME_OWNER_MAP,
+    "remarkable": OS_TO_REMARKABLE_AMI_NAME_OWNER_MAP,
+    "pcluster": OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP,
+}
 
 
 def retrieve_latest_ami(region, os, ami_type="official", architecture="x86_64"):
-    ec2_client = boto3.client("ec2", region_name=region)
-    response = ec2_client.describe_images(
-        Filters=[
-            {"Name": "name", "Values": [AMI_TYPE_DICT[ami_type][os]["name"]]},
-            {"Name": "architecture", "Values": [architecture]},
-        ],
-        Owners=AMI_TYPE_DICT[ami_type][os]["owners"],
-    )
-    # Sort on Creation date Desc
-    amis = sorted(response["Images"], key=lambda x: x["CreationDate"], reverse=True)
-    return amis[0]["ImageId"]
+    try:
+        if ami_type == "pcluster":
+            ami_name = "aws-parallelcluster-{version}-{ami_name}".format(
+                version=get_installed_version(), ami_name=AMI_TYPE_DICT.get(ami_type).get(os).get("name")
+            )
+        else:
+            ami_name = AMI_TYPE_DICT.get(ami_type).get(os).get("name")
+        response = boto3.client("ec2", region_name=region).describe_images(
+            Filters=[{"Name": "name", "Values": [ami_name]}, {"Name": "architecture", "Values": [architecture]}],
+            Owners=AMI_TYPE_DICT.get(ami_type).get(os).get("owners"),
+        )
+        # Sort on Creation date Desc
+        amis = sorted(response["Images"], key=lambda x: x["CreationDate"], reverse=True)
+        return amis[0]["ImageId"]
```

Comment:
If there are no images in the list, the `amis[0]` can raise an `IndexError` or if there is no `ImageId` (unlikely) this can raise a `KeyError`.

Reply:
Done